### DISCUSSION TITLE
Log rotation

### DISCUSSION THREAD

**Discussion by absey22** [Koenkk/zigbee2mqtt#7360] [General]
Hi,

I am using zigbee2mqtt with a few CC2531 chips as coordinator and end devices. My application requires that my logging (in Linux, the files in /opt/zigbee2mqtt/data/) must last for multiple days consecutively. To this end, I am using the log_rotation feature.

However, I have noticed that the rotation of file names goes only up to 2 (that is, log.txt, then log1.txt, then log2.txt, then back to log.txt) And soI start losing my earlier data stored in log.txt.

As I would like to keep the logging running for several days (requiring at least 5 or 6 files, of greater than 9MB each) I need the log rotation to extend up to log3.txt, log4.txt,... etc.

Is there an additional setting somewhere I can enable to ensure that the log rotation continue for a longer period before reverting back to overwriting log.txt? Or should I make some sort of bash script to make sure I copy the files periodically before they are overwritten by the log rotation?

Any help is much appreciated,
Aaron

**Comment by florentin68** [comment]
You should use `logrotate` and add a configuration file like this: `/etc/logrotate.d/zigbee2mqtt`.
I have this for example:
```
/var/log/zigbee2mqtt/log.txt {
        rotate 20
        weekly
#       compress
#       delaycompress
#       size 1M
        create 0644 www-data users
        missingok
        notifempty
        prerotate
                systemctl stop zigbee2mqtt
        endscript
        postrotate
                systemctl start zigbee2mqtt
        endscript
}
```

I think `rotate 20` tells logrotate to keep 20 files, 20 weeks (`weekly`) for me then.
[Logrotate manpage](https://linux.die.net/man/8/logrotate)

**Reply by absey22** [reply]
Thanks so much! This is exactly what I needed.
